components/script/dom/element.rs

```diff
@@ -363,6 +363,19 @@ impl Element {
         !self.overflow_y_is_visible()
     }
 
+    // https://drafts.csswg.org/cssom-view/#scrolling-box
+    fn has_scrolling_box(&self) -> bool {
+        // TODO: scrolling mechanism, such as scrollbar (We don't have scrollbar yet)
+        //       self.has_scrolling_mechanism()
+        self.overflow_x_is_hidden() ||
```

Comment:
Hmm, there's a big condition that is missing here, which is:

> it overflows its content area and the used value of the overflow-x or overflow-y property is hidden.

So, is there something missing here? If not, where does it get handled? If nowhere, should it? Or should at least be a `TODO` here?

Reply:
In my thought, when overflow-x or overflow-y property is `hidden`, it must `it overflows its content area`.
Because, when it is not `hidden`, it cannot overflow anymore.

Reply:
That's not true, an element can definitely overflow its content area with overflow other than `hidden`.

See for example:

```
<div style="width: 100px; height: 100px;">
    <div>Something that takes more than 100px</div>
</div>
```

But in general the spec doesn't make much sense to me, it seems to me that the element would have a scroll box when the content overflows and the `overflow` property is _not_ `hidden`... I cannot make any sense of the spec logic here, maybe @jdm or @SimonSapin can double-check me?

Reply:
Let me take a closer look at the test and other implementations...

Reply:
Ok, I see the intent now, and the "element overflows its content area" bit is pretty important.

For example, in your test, even if `overflow` is `hidden`, there's no scrolling box when `#scrollable`s width and height is less than `#section1`s size.

A test should probably be added for this, and this is basically the `has_overflow` check if I understand correctly. So you can put it here, I think.

Reply:
I thought `element overflows its content area` and `has overflow` is equal?
The fourth test has tested `has_overflow`.
I am not sure if I got your points.

Reply:
@emilio  Shall I change anything?